From 3d06bf152abcc3895a0f3afa21d762d84c9aecbc Mon Sep 17 00:00:00 2001
From: Oliver Neukum <oliver@neukum.org>
Date: Thu, 10 Feb 2011 15:33:17 +0100
Subject: usb_wwan: fix runtime PM in error case
Patch-mainline: v2.6.39
Git-commit: 3d06bf152abcc3895a0f3afa21d762d84c9aecbc

An error in the write code path would permanently disable
runtime PM in this driver

Signed-off-by: Oliver Neukum <oliver@neukum.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/usb_wwan.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/usb/serial/usb_wwan.c
+++ b/drivers/usb/serial/usb_wwan.c
@@ -261,6 +261,7 @@ int usb_wwan_write(struct tty_struct *tt
 				intfdata->in_flight--;
 				spin_unlock_irqrestore(&intfdata->susp_lock,
 						       flags);
+				usb_autopm_put_interface_async(port->serial->interface);
 				continue;
 			}
 		}
